core/src/main/scala/fs2/job/JobManager.scala

```diff
@@ -165,7 +166,8 @@ final class JobManager[F[_]: Concurrent: Timer, I, N] private (
               if (ignoreAbsence) {
                 val casF = Concurrent[F] delay {
                   val attempt = Context(Status.Running, Some(s.set(true)))
-                  attempt eq meta.putIfAbsent(id, attempt)
+
+                  Option(meta.putIfAbsent(id, attempt)).isEmpty
```

Comment:
`putIfAbsent` returns null and mutates `meta` when `id` is absent. The previous implementation always yielded false.